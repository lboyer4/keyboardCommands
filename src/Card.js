import React, { Component } from 'react';
import './Card.scss';

class Card extends Component {
	constructor() {
		super()



	}
  render() {
    return (
    
    );
  }
}

export default Card