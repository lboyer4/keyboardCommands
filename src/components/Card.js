import React, { Component } from 'react';
import '../SCSS/_Card.scss';

class Card extends Component {
	constructor() {
		super()
		this.state = {
			answered: false,
			nextCard: false,
		}
	}

	// handleAnswer = (e) => {
	// 	e.preventDefault()
	// 	console.log(e.target.parentElement.querySelector('.answer').value)
	// 	let playerAnswer = e.target.parentElement.querySelector('.answer').value
	// 	// console.log(this.props.command)
	// }

  render() {
  	let nextCard = <button className="next" onClick={this.changeCard}>Next</button>
  	// let nextButton = (this.state.answered) ? nextCard 
    return (
    	<article className="card">
    		<form>
	    		<h4>{this.props.task}</h4>
	    		<input className="answer" type="text" placeholder="input answer here..." />
	    		<h5>{this.props.command}</h5>
	    		<button className="answer-btn" onClick={this.handleAnswer}>Submit Answer</button>
	    		{/*{nextButton}*/}
    		</form>
    	</article>
    
    )
  }
}

export default Card;