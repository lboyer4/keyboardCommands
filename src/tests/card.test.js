import React from 'react';
import ReactDOM from 'react-dom';
import Card from '../components/Card';
import { shallow } from 'enzyme'; 

describe("Card", () => {
	let wrapper;


	const mocksetIncorrectStack = jest.fn();
	const mockUpdateScore = jest.fn();
	const mocksetWrongCards = jest.fn();
	const mocksetCurrentCard = jest.fn()
	// const mockSetCurrentCard = jest.fn();

	beforeEach(() => {
		wrapper = shallow(
					<Card 
					updateScore={mockUpdateScore}
					setIncorrectStack={mocksetIncorrectStack}
					setWrongCards={mocksetWrongCards}
					setCurrentCard={mocksetCurrentCard}
					task={"Save"}
					command={"command S"}
					level={'Beginner'}
					incorrectCards={[]}/>
		)
	});

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});


	it('should compare answers when clicked', () => {
		wrapper.state().answered=false;
		wrapper.find('.answer-btn').simulate('click');
		expect(wrapper.state().answered).toEqual(true);
	})

	// it('should call updateScores on submit', () => {
 //    wrapper.find('.answer-btn').simulate('click', { preventDefault: () => {}
 //  });
 //  //   const wrapperTwo = mount(<Card />);
	// 	// expect(wrapperTwo.find('.answer').parents()).toEqual('command');
 //    expect(mockUpdateScore).toBeCalled();
 //  });	

	 // it('should return answered to false when called', () => {

  // });

	 // it('set the wrong cards when clicked', () => {
	 // 	wrapper.find('.wrong-cards-btn').simulate('click');
	 // 	expect(mocksetWrongCards).toBeCalled();
	 // })
});