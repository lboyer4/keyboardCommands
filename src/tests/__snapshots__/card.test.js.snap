// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<article
  className="card"
>
  <form>
    <h4>
      Save
    </h4>
    <input
      className="answer"
      placeholder="input answer here..."
      type="text"
    />
    <button
      className="answer-btn"
      onClick={[Function]}
    >
      Submit Answer
    </button>
  </form>
</article>
`;
